metricbeat/module/windows/service/service_status.go

```diff
@@ -168,7 +168,8 @@ func GetServiceStates(handle Handle, state ServiceEnumState, protectedServices m
 
 		service, err := getServiceInformation(serviceTemp, servicesBuffer, handle, protectedServices)
 		if err != nil {
-			return nil, err
+			logp.Err("error while getting service information: %v", err)
```

Comment:
I'm skeptical about the `error` log level here. This will be logged on each iteration when we collect metrics and on each service that does not allow access.

We have to use the debug level here and re-phrase the message.

```suggestion
			logp.Debug("could not get information for the service: %v", err)
```

Is it possible to also propagate and include the service name here?

Reply:
> I'm skeptical about the error log level here. This will be logged on each iteration when we collect metrics and on each service that does not allow access.

No. This will log only if the error is _**not**_ access related.

If you see here, https://github.com/elastic/beats/blob/7ee312854507192d2b3b8eff315cc23fac7d79f4/metricbeat/module/windows/service/service_status.go#L238-L241
We return error if it is _**not**_ access related. Or else, we log it only once.

Reply:
maybe we could make it an error but ensure it only appears once per service pid?

Reply:
> Is it possible to also propagate and include the service name here?

It is, but if the service collection has failed while fetching other information then we could have empty document on ES. I think its better to log for now. wdyt?

Reply:
> maybe we could make it an error but ensure it only appears once per service pid?

That's doable. Would we do it in a follow-up or in this PR?

Reply:
> It is, but if the service collection has failed while fetching other information then we could have empty document on ES. I think its better to log for now. wdyt?

not sure I understand this statement. Why would we send an empty document to ES? We just need to communicate to the user in logs which service Metricbeat is not able to read, I think it's essential for troubleshooting.

Reply:
Denis, I misunderstood your original comment. I'll update the log to include the service name.